addons/pos_adyen/views/pos_payment_provider_views.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="pos_payment_provider_view_form_inherit_pos_adyen" model="ir.ui.view">
+        <field name="name">pos.payment.provider.form.inherit.adyen</field>
+        <field name="model">pos.payment.provider</field>
+        <field name="inherit_id" ref="point_of_sale.pos_payment_provider_view_form"/>
+        <field name="arch" type="xml">
+            <group name="provider_credentials" position='inside'>
+                <group name="adyen_credentials" invisible="code != 'adyen'">
+                       <field name="adyen_api_key" required="code == 'adyen' and state != 'disabled'" password="True"/>
```

Comment:
for other keys in other payment providers, maybe it's a good idea to make them `password="True"` just like we do here in `pos_adyen`.